autoretrieve/autoretrieve.go

```diff
@@ -38,12 +33,33 @@ type Autoretrieve struct {
 	Addresses         string
 }
 
+func (autoretrieve *Autoretrieve) AddrInfo() (*peer.AddrInfo, error) {
```

Comment:
should we be reusing the logic to construct the full AddrInfo here instead of just taking the first address? https://github.com/application-research/estuary/pull/510/files#diff-56307bdb7ded67ebe516f562b90384407b21b2d34e8b5766b42602e1fb9705f0R4777-R4794

Reply:
yes! thanks for linking the place where it was done before, i will fix

Reply:
omg thank you thank you for finding this, after you pointed this out i realized the published advertisements were not including the peer ID. it's been fixed, you saved that day! <3

Reply:
was also able to de-dup the address info parsing

Reply:
as it turns out, advertisements should not include peer id in the multiaddr. but the code dedup is still a good thing : )